config/main.py

```diff
@@ -1313,5 +1315,281 @@ def naming_mode_alias():
+#
+@interface.command()
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.pass_context
+def enable(ctx, ifname):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
+        click.echo("Invalid interface name")
+        return
+
+    config_db = ctx.obj['db']
+    intf_dict = config_db.get_table('SFLOW_SESSION')
+
+    if intf_dict and ifname in intf_dict.keys():
+        intf_dict[ifname]['admin_state'] = 'enable'
+        config_db.set_entry('SFLOW_SESSION', ifname, intf_dict[ifname])
+    else:
+        config_db.set_entry('SFLOW_SESSION', ifname, {'admin_state': 'enable'})
+
+#
+# 'sflow' command ('config sflow interface disable  ...')
+#
+@interface.command()
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.pass_context
+def disable(ctx, ifname):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
+        click.echo("Invalid interface name")
+        return
+
+    config_db = ctx.obj['db']
+    intf_dict = config_db.get_table('SFLOW_SESSION')
+
+    if intf_dict and ifname in intf_dict.keys():
+        intf_dict[ifname]['admin_state'] = 'disable'
+        config_db.set_entry('SFLOW_SESSION', ifname, intf_dict[ifname])
+    else:
+        config_db.set_entry('SFLOW_SESSION', ifname,
+                            {'admin_state': 'disable'})
+
+#
+# 'sflow' command ('config sflow interface sample-rate  ...')
+#
+@interface.command('sample-rate')
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.argument('rate', metavar='<sample_rate>', required=True, type=int)
+@click.pass_context
+def sample_rate(ctx, ifname, rate):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
+        click.echo('Invalid interface name')
+        return
+    if not is_valid_sample_rate(rate):
+        click.echo('Error: Sample rate must be between 256 and 8388608')
+        return
+
+    config_db = ctx.obj['db']
+    sess_dict = config_db.get_table('SFLOW_SESSION')
+
+    if sess_dict and ifname in sess_dict.keys():
+        sess_dict[ifname]['sample_rate'] = rate
+        config_db.set_entry('SFLOW_SESSION', ifname, sess_dict[ifname])
+    else:
+        config_db.set_entry('SFLOW_SESSION', ifname, {'sample_rate': rate})
+
+
+#
+# 'sflow collector' group
+#
+@sflow.group()
+@click.pass_context
+def collector(ctx):
+    """Add/Delete a sFlow collector"""
+    pass
+
+#
+# 'sflow' command ('config sflow collector add ...')
+#
+@collector.command()
+@click.option('--port', required=False, type=int, default=6343,
+              help='Collector port number')
+@click.argument('name', metavar='<collector_name>', required=True)
+@click.argument('ipaddr', metavar='<IPv4/v6_address>', required=True)
+@click.pass_context
+def add(ctx, name, ipaddr, port):
+    """Add a sFlow collector"""
+    ipaddr = ipaddr.lower()
+
+    if not is_valid_collector_info(name, ipaddr, port):
```

Comment:
Do you see an issue with function being defined after invocation? I see that this `is_valid_collector_info` function is defined below and not above this.

Reply:
Didn't notice a problem, but will move it up.